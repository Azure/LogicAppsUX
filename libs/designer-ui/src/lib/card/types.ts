export const MenuItemType = {
  Normal: 0,
  Divider: 1,
  Header: 2,
  Advanced: 3,
} as const;
export type MenuItemType = (typeof MenuItemType)[keyof typeof MenuItemType];

export interface CommentBoxProps {
  brandColor: string;
  comment: string;
  isDismissed: boolean;
  isEditing: boolean;
  isPanelModeEnabled?: boolean;
  styleWidth?: string;
  onCommentChanged?(e: CommentChangeEvent): void;
  onCommentCommitted?(): void;
  onCommentDismissed?(): void;
}

interface CommentChangeEvent {
  value: string;
}
