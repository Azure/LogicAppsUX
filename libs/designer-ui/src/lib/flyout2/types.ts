export interface FlyoutSelectedEventArgs {
  key: string;
}

export type FlyoutSelectedEventHandler = (e: FlyoutSelectedEventArgs) => void;
