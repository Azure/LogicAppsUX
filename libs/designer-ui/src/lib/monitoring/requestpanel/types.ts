import type { ContentLink, SecureData } from '@microsoft/logic-apps-shared';

export interface RequestHistory {
  properties: RequestHistoryProperties;
}

export interface RequestHistoryRequest {
  body?: any;
  bodyLink?: ContentLink;
  headers: Record<string, string>;
  method: string;
  uri: string;
}

export interface RequestHistoryResponse {
  body?: any;
  bodyLink?: ContentLink;
  headers: Record<string, string>;
  statusCode: number;
}

interface ExtendedErrorInfo {
  code: string;
  details?: ExtendedErrorInfo[];
  innerError?: any;
  message: string;
}

interface RequestHistoryProperties {
  endTime?: string;
  error?: ExtendedErrorInfo;
  request?: RequestHistoryRequest;
  response?: RequestHistoryResponse;
  startTime: string;
  secureData?: SecureData;
}
