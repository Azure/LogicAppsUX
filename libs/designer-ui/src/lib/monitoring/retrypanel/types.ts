export interface RetryHistory {
  clientRequestId: string;
  code: string;
  endTime?: string;
  error?: any;
  serviceRequestId?: string;
  startTime: string;
}
