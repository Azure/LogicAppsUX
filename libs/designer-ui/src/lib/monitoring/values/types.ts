export interface ContentHash {
  algorithm: string;
  value: string;
}

export interface ContentLink {
  contentHash: ContentHash;
  contentSize: number;
  contentVersion: string;
  metadata?: Record<string, unknown>;
  secureData?: SecureData;
  uri?: string;
}

export interface SecureData {
  properties: string[];
}

export interface ValueProps {
  displayName: string;
  format?: string;
  language?: string;
  value: any;
  visible?: boolean;
}

export interface Xml {
  '$content-type': string;
  $content: string;
}
