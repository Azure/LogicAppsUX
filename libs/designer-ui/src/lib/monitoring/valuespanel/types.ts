export type BoundParameters = Record<string, BoundParameter<any>>;

interface BoundParameter<T> {
  displayName: string;
  dynamicValue?: T;
  format?: string;
  language?: string;
  value: T;
  visibility?: string;
}
