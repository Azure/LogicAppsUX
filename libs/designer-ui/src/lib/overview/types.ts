import { isObject } from '@microsoft-logic-apps/utils';

export type CallbackInfo = CallbackInfoWithRelativePath | CallbackInfoWithValue;

export interface CallbackInfoWithRelativePath {
  basePath?: string;
  method: string;
  relativeParameters?: unknown[];
  relativePath: string;
  queries?: Record<string, string>;
}

export interface CallbackInfoWithValue {
  value: string;
}

export interface Run {
  id: string;
  name: string;
  properties: RunProperties;
  kind?: string;
  location?: string;
  tags?: string;
  type: string;
}

export interface RunDisplayItem {
  duration: string;
  id: string;
  identifier: string;
  startTime: string;
  status: string;
}

export interface RunError {
  error: {
    code: string;
    message: string;
  };
}

export interface RunProperties {
  actions?: Record<string, unknown>;
  code?: string;
  correlation?: unknown;
  correlationId?: string;
  endTime?: string;
  error?: RunError;
  outputs: Record<string, unknown>;
  startTime: string;
  status: string;
  trigger: unknown;
  workflow: unknown;
}

export interface Runs {
  nextLink?: string;
  runs: Run[];
}

export function isCallbackInfoWithRelativePath(value: any): value is CallbackInfoWithRelativePath {
  return isObject(value) && typeof value.relativePath === 'string';
}
