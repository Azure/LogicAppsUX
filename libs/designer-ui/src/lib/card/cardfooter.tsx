/* eslint-disable react/display-name */
import type { CardProps } from './index';
import { getHeaderStyle } from './utils';
import type { IIconProps } from '@fluentui/react';
import { css, Icon } from '@fluentui/react';
import { Tooltip } from '@fluentui/react-components';
import { memo, useMemo } from 'react';
import { useIntl } from 'react-intl';

export type CardFooterProps = Pick<
  CardProps,
  'commentBox' | 'connectionDisplayName' | 'connectionRequired' | 'staticResultsEnabled' | 'isSecureInputsOutputs' | 'nodeIndex'
>;

interface CardBadgeBarProps {
  badges: CardBadgeProps[];
  brandColor?: string;
  tabIndex?: number;
}

interface CardBadgeProps {
  active: boolean;
  content: string;
  darkBackground?: boolean;
  iconProps: IIconProps;
  title: string;
  tabIndex?: number;
}

const commentIconProps: IIconProps = {
  iconName: 'Comment',
};

const connectionIconProps: IIconProps = {
  iconName: 'Link',
};

const staticResultIconProps: IIconProps = {
  iconName: 'TestBeaker',
};

const lockIconProps: IIconProps = {
  iconName: 'Lock',
};

export const CardFooter: React.FC<CardFooterProps> = memo(
  ({ commentBox, connectionDisplayName, connectionRequired, staticResultsEnabled, isSecureInputsOutputs, nodeIndex }) => {
    const intl = useIntl();

    const strings = useMemo(
      () => ({
        CONNECTION_NAME_DISPLAY: intl.formatMessage({
          defaultMessage: 'Connection name',
          id: 'XOzn/3',
          description: 'This is for a label for a badge, it is used for screen readers and not shown on the screen.',
        }),
        CONNECTION_CONTAINER_CONNECTION_REQUIRED: intl.formatMessage({
          defaultMessage: 'Connection required',
          id: 'CG772M',
          description: 'This is for a label for a badge, it is used for screen readers and not shown on the screen.',
        }),
        PANEL_STATIC_RESULT_TITLE: intl.formatMessage({
          defaultMessage: 'Testing',
          id: 'm7Y6Qf',
          description: 'Title for a tab panel',
        }),
        MENU_STATIC_RESULT_ICON_TOOLTIP: intl.formatMessage({
          defaultMessage: 'This action has testing configured.',
          id: 'WxcmZr',
          description: "This is a tooltip for the Status results badge shown on a card. It's shown when the baged is hovered over.",
        }),
        COMMENT: intl.formatMessage({
          defaultMessage: 'Comment',
          id: 'VXBWrq',
          description: 'This is for a label for a badge, it is used for screen readers and not shown on the screen.',
        }),
        SECURE_INPUTS_OUTPUTS_TITLE: intl.formatMessage({
          defaultMessage: 'Secure inputs or outputs enabled.',
          id: '0F6jmK',
          description: 'Secure inputs or outputs enabled.',
        }),
        SECURE_INPUTS_OUTPUTS_TOOLTIP: intl.formatMessage({
          defaultMessage: 'This operation has secure inputs or outputs enabled.',
          id: 'byRkj+',
          description: 'This operation has secure inputs or outputs enabled.',
        }),
      }),
      [intl]
    );

    const connectionTitle = useMemo(
      () => (connectionDisplayName ? strings.CONNECTION_NAME_DISPLAY : strings.CONNECTION_CONTAINER_CONNECTION_REQUIRED),
      [connectionDisplayName, strings]
    );

    const badges: CardBadgeProps[] = useMemo(
      () => [
        ...(staticResultsEnabled
          ? [
              {
                active: true,
                content: strings.MENU_STATIC_RESULT_ICON_TOOLTIP,
                iconProps: staticResultIconProps,
                title: strings.PANEL_STATIC_RESULT_TITLE,
              },
            ]
          : []),
        ...(commentBox && commentBox.comment
          ? [
              {
                active: true,
                content: commentBox.comment,
                iconProps: commentIconProps,
                title: strings.COMMENT,
              },
            ]
          : []),
        ...(connectionRequired || connectionDisplayName
          ? [
              {
                active: !!connectionDisplayName,
                content: connectionDisplayName || connectionTitle,
                iconProps: connectionIconProps,
                title: connectionTitle,
              },
            ]
          : []),
        ...(isSecureInputsOutputs
          ? [
              {
                active: true,
                content: strings.SECURE_INPUTS_OUTPUTS_TOOLTIP,
                iconProps: lockIconProps,
                title: strings.SECURE_INPUTS_OUTPUTS_TITLE,
              },
            ]
          : []),
      ],
      [commentBox, connectionDisplayName, connectionRequired, connectionTitle, isSecureInputsOutputs, staticResultsEnabled, strings]
    );

    if (!badges.length) {
      return null;
    }

    return (
      <div className="msla-card-v2-footer">
        <CardBadgeBar badges={badges} tabIndex={nodeIndex} />
      </div>
    );
  }
);

const CardBadgeBar: React.FC<CardBadgeBarProps> = ({ badges, brandColor, tabIndex }) => {
  return (
    <div className="msla-badges" style={getHeaderStyle(brandColor)}>
      {badges.map(({ active, content, darkBackground, iconProps, title }) => (
        <CardBadge
          key={title}
          title={title}
          content={content}
          darkBackground={darkBackground}
          iconProps={iconProps}
          active={active}
          tabIndex={tabIndex}
        />
      ))}
    </div>
  );
};

const CardBadge: React.FC<CardBadgeProps> = ({ active, content, darkBackground = false, iconProps, title, tabIndex }) => {
  if (!content) {
    return null;
  }
  if (active) {
    return (
      <Tooltip relationship={'label'} withArrow={true} content={content}>
        <div>
          <Icon
            className={css('panel-card-v2-badge', 'active', darkBackground && 'darkBackground')}
            {...iconProps}
            ariaLabel={`${title}: ${content}`}
            tabIndex={tabIndex}
          />
        </div>
      </Tooltip>
    );
  }
  return <Icon className="panel-card-v2-badge inactive" {...iconProps} ariaLabel={title} tabIndex={0} />;
};
