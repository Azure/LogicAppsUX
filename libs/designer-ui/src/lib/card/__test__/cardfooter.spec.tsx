import renderer from 'react-test-renderer';
import { CardFooter, CardFooterProps } from '../cardfooter';
import { CommentBoxProps } from '../types';

describe('lib/card/errorbanner', () => {
  let minimal: CardFooterProps;

  beforeEach(() => {
    minimal = {};
  });

  it('should render', () => {
    const tree = renderer.create(<CardFooter {...minimal} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with a comment icon', () => {
    const commentBox: CommentBoxProps = {
      brandColor: '#474747',
      comment: 'comment',
      isDismissed: false,
      isEditing: false,
    };
    const tree = renderer.create(<CardFooter {...minimal} commentBox={commentBox} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with a connection icon', () => {
    const tree = renderer
      .create(<CardFooter {...minimal} connectionDisplayName="joechung@microsoft.com" connectionRequired={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with an inactive connection icon', () => {
    const tree = renderer.create(<CardFooter {...minimal} connectionDisplayName="" connectionRequired={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with a testing icon', () => {
    const tree = renderer.create(<CardFooter {...minimal} staticResultsEnabled={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
