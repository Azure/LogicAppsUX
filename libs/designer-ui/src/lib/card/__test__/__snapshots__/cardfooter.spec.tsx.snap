// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/card/cardfooter should render 1`] = `
<div
  className="msla-card-v2-footer"
>
  <div
    className="msla-badges"
  />
</div>
`;

exports[`lib/card/cardfooter should render with a comment icon 1`] = `
<div
  className="msla-card-v2-footer"
>
  <div
    className="msla-badges"
  >
    <div
      aria-label="comment"
      onBlur={[Function]}
      onFocus={[Function]}
      onPointerEnter={[Function]}
      onPointerLeave={[Function]}
    >
      <i
        aria-label="Comment: comment"
        className="panel-card-v2-badge active root-111"
        data-icon-name="Comment"
        role="img"
        tabIndex={0}
      >
        
      </i>
    </div>
  </div>
</div>
`;

exports[`lib/card/cardfooter should render with a connection icon 1`] = `
<div
  className="msla-card-v2-footer"
>
  <div
    className="msla-badges"
  >
    <div
      aria-label="joechung@microsoft.com"
      onBlur={[Function]}
      onFocus={[Function]}
      onPointerEnter={[Function]}
      onPointerLeave={[Function]}
    >
      <i
        aria-label="Connection name: joechung@microsoft.com"
        className="panel-card-v2-badge active root-114"
        data-icon-name="Link"
        role="img"
        tabIndex={0}
      >
        
      </i>
    </div>
  </div>
</div>
`;

exports[`lib/card/cardfooter should render with a lock icon 1`] = `
<div
  className="msla-card-v2-footer"
>
  <div
    className="msla-badges"
  >
    <div
      aria-label="This operation has secure inputs or outputs enabled."
      onBlur={[Function]}
      onFocus={[Function]}
      onPointerEnter={[Function]}
      onPointerLeave={[Function]}
    >
      <i
        aria-label="Secure inputs or outputs enabled.: This operation has secure inputs or outputs enabled."
        className="panel-card-v2-badge active root-114"
        data-icon-name="Lock"
        role="img"
        tabIndex={0}
      >
        
      </i>
    </div>
  </div>
</div>
`;

exports[`lib/card/cardfooter should render with a testing icon 1`] = `
<div
  className="msla-card-v2-footer"
>
  <div
    className="msla-badges"
  >
    <div
      aria-label="This Action has testing configured."
      onBlur={[Function]}
      onFocus={[Function]}
      onPointerEnter={[Function]}
      onPointerLeave={[Function]}
    >
      <i
        aria-label="Testing: This Action has testing configured."
        className="panel-card-v2-badge active root-117"
        data-icon-name="TestBeaker"
        role="img"
        tabIndex={0}
      >
        
      </i>
    </div>
  </div>
</div>
`;

exports[`lib/card/cardfooter should render with an inactive connection icon 1`] = `
<div
  className="msla-card-v2-footer"
>
  <div
    className="msla-badges"
  >
    <i
      aria-label="Connection required"
      className="panel-card-v2-badge inactive root-114"
      data-icon-name="Link"
      role="img"
      tabIndex={0}
    >
      
    </i>
  </div>
</div>
`;
